You will be provided with a partial code base and an issue statement explaining a problem to resolve.

<issue>
Support Replicated Database in ClickHouse Copier
**Use case**

ClickHouse copier doesn't currently work with replicated databases. It results in the following error:

```
2022.03.09 13:55:03.267739 [ 2160190 ] {} <Error> Application: Code: 80. DB::Exception: Received from k8s-ns179cef-c179cef1-da30ec77d6-0ae794c2c59d4061.elb.us-east-2.amazonaws.com:9440. DB::Exception: It's not initial query. ON CLUSTER is not allowed for Replicated database.. Stack trace:

0. DB::Exception::Exception(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, int, bool) @ 0xb06a5ba in /usr/bin/clickhouse
1. DB::DatabaseReplicated::tryEnqueueReplicatedDDL(std::__1::shared_ptr<DB::IAST> const&, std::__1::shared_ptr<DB::Context const>) @ 0x14f76371 in /usr/bin/clickhouse
2. DB::InterpreterCreateQuery::createTable(DB::ASTCreateQuery&) @ 0x15625b65 in /usr/bin/clickhouse
3. DB::InterpreterCreateQuery::execute() @ 0x1562d44b in /usr/bin/clickhouse
4. ? @ 0x15959c29 in /usr/bin/clickhouse
5. DB::executeQuery(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, std::__1::shared_ptr<DB::Context>, bool, DB::QueryProcessingStage::Enum) @ 0x15957775 in /usr/bin/clickhouse
6. DB::TCPHandler::runImpl() @ 0x1649ddfa in /usr/bin/clickhouse
7. DB::TCPHandler::run() @ 0x164b1d99 in /usr/bin/clickhouse
8. Poco::Net::TCPServerConnection::start() @ 0x19768a4f in /usr/bin/clickhouse
9. Poco::Net::TCPServerDispatcher::run() @ 0x1976aea1 in /usr/bin/clickhouse
10. Poco::PooledThread::run() @ 0x19927ea9 in /usr/bin/clickhouse
11. Poco::ThreadImpl::runnableEntry(void*) @ 0x19925200 in /usr/bin/clickhouse
12. ? @ 0x7f9f93792609 in ?
13. clone @ 0x7f9f936b9293 in ?
. (INCORRECT_QUERY), Stack trace (when copying this message, always include the lines below):
```

**Describe the solution you'd like**

Remove `ON_CLUSTER` for  replicated db.

DatabaseReplicated: Replica ... already exists. Replica host ID: 'DROPPED'
https://s3.amazonaws.com/clickhouse-test-reports/33847/8921725d716e05b13dc82f2db1c3d5bfc3ad6983/stress_test__debug__actions_.html
Always use default database when creating distributed table
**Describe the unexpected behaviour**
Always use default database when creating distributed table

**How to reproduce**
* Which ClickHouse server version to use
22.4.5.9

The following statements fail with exceptions:

```
create database test on cluster test_cluster;

use test;

create table local_t_l5ydey on cluster test_cluster ( 
c_qv5rv INTEGER ,
c_rutjs4 INTEGER ,
c_wmj INTEGER ,
c_m3 TEXT NOT NULL,
primary key(c_qv5rv)
)
engine=ReplicatedMergeTree('/clickhouse/tables/test/{shard}/local_t_l5ydey', '{replica}');

create table t_l5ydey on cluster test_cluster as local_t_l5ydey
engine=Distributed('test_cluster','test','local_t_l5ydey', rand())
```

```
Query id: c0de0142-df6f-4703-9a68-9a491263133d

┌─host──────┬─port─┬─status─┬─error────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┬─num_hosts_remaining─┬─num_hosts_active─┐
│ 127.0.0.1 │ 5000 │     60 │ Code: 60. DB::Exception: Table default.local_t_l5ydey doesn't exist. (UNKNOWN_TABLE) (version 22.4.5.9 (official build)) │                   3 │                0 │
│ 127.0.0.1 │ 5001 │     60 │ Code: 60. DB::Exception: Table default.local_t_l5ydey doesn't exist. (UNKNOWN_TABLE) (version 22.4.5.9 (official build)) │                   2 │                0 │
│ 127.0.0.1 │ 5002 │     60 │ Code: 60. DB::Exception: Table default.local_t_l5ydey doesn't exist. (UNKNOWN_TABLE) (version 22.4.5.9 (official build)) │                   1 │                0 │
│ 127.0.0.1 │ 5003 │     60 │ Code: 60. DB::Exception: Table default.local_t_l5ydey doesn't exist. (UNKNOWN_TABLE) (version 22.4.5.9 (official build)) │                   0 │                0 │
└───────────┴──────┴────────┴──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┴─────────────────────┴──────────────────┘
→ Progress: 0.00 rows, 0.00 B (0.00 rows/s., 0.00 B/s.)  0%
4 rows in set. Elapsed: 0.173 sec. 

Received exception from server (version 22.4.5):
Code: 60. DB::Exception: Received from localhost:5000. DB::Exception: There was an error on [127.0.0.1:5000]: Code: 60. DB::Exception: Table default.local_t_l5ydey doesn't exist. (UNKNOWN_TABLE) (version 22.4.5.9 (official build)). (UNKNOWN_TABLE)
```

The "create table ... as ..." statement uses the default database after the "use" statement.

**Expected behavior**
The "create table ... as ..." statement should use the current used database.
</issue>

I need you to solve the provided issue by generating a code fix that can be applied directly to the repository

Respond below:
